crates/net/network/src/session/active.rs

```diff
@@ -322,9 +323,27 @@ impl ActiveSession {
         for id in timedout {
             warn!(target: "net::session", ?id, remote_peer_id=?self.remote_peer_id, "timed out outgoing request");
             let req = self.inflight_requests.remove(&id).expect("exists; qed");
+            self.update_request_timeout(req.timestamp, req.deadline);
             req.request.send_err_response(RequestError::Timeout);
         }
     }
+
+    /// Updates the request timeout with a message's timestamps
+    fn update_request_timeout(&mut self, sent: Instant, received: Instant) {
+        // TODO: inline or constant? where should we put this? What would be
+        // a sensible value for the minimum?
+        /// Minimum timeout value
+        const MINIMUM_TIMEOUT: Duration = Duration::from_millis(1);
+        /// Amount of RTTs before timeout
+        const TIMER_MULTIPLIER: u32 = 4;
+
+        // TODO: use a more sophisticated formula
```

Comment:
worth having a look how geth does this here

Reply:
Just checked. They use [the same formula](https://github.com/ethereum/go-ethereum/blob/2c6dda5ad7a720cccd957230f7978de0082ec8c7/p2p/msgrate/msgrate.go#L198).